src/profilers.cpp

```diff
@@ -2,19 +2,373 @@
+
+      while (command_status > 0){
+
+        struct timespec start, end;
+        clock_gettime(CLOCK_MONOTONIC, &start);
+
+        int pipe_fd[2];
+
+        if (pipe(pipe_fd) == -1) {
+          print("Could not establish communication pipe between metric-exec "
+                "and metric-read in \"" + this->get_name() + "\"! Terminating "
+                "the profiled command wrapper.", true, true);
+          kill(pid, SIGTERM);
+          return;
+        }
+
+        pid_t forked_metric_exec = fork();
+        if (forked_metric_exec == 0){
+          close(pipe_fd[0]);
+
+          fs::current_path(result_processed);
+
+          int stderr_fd = creat(stderr_metric_command.c_str(), O_WRONLY);
+
+          if (stderr_fd == -1) {
+            std::exit(ERROR_STDERR);
+          }
+
+          if (dup2(stderr_fd, STDERR_FILENO) == -1) {
+            std::exit(ERROR_STDERR_DUP2);
+          }
+
+          if (dup2(pipe_fd[1], STDOUT_FILENO) == -1) {
+            std::exit(ERROR_STDOUT_DUP2);
+          }
+
+          close(pipe_fd[1]);
+          execvp(path, parts_c_str);
+
+          std::exit(errno);
+        }
+        else
+        {
+          close(pipe_fd[1]);
+          //read
+          char buffer[1024];
+          ssize_t bytes_read;
+          std::string data;
+          
+          while ((bytes_read = read(pipe_fd[0], buffer, sizeof(buffer) - 1)) > 0) {
+              buffer[bytes_read] = '\0';
+              data += buffer;
+          }
+          std::string parsed_data;
+
+          if(!this->regex.empty()){
+
+            std::regex regexPattern(this->regex);
+            std::smatch match_pattern;
+            if (std::regex_search(data, match_pattern, regexPattern)) {
+                parsed_data = match_pattern.str();
+            }else{
+                std::exit(ERROR_USER_REGEX_MATCH);
+            }
+
+          }else{
+            parsed_data = data;
+          }
+
+          std::regex number_regex(R"(-?\d+(\.\d+)?)");;
+
+          std::smatch match;
+          std::string temp_data = parsed_data;
+          int count = 0;
+          float metric_val;
+                    
+          while (std::regex_search(temp_data, match, number_regex)) {
+            count++;
+            if(count > 1){break;}
+  
+            std::string number_str = match.str(0);
+              try {
+                  metric_val = std::stof(number_str);  
+              } catch (const std::exception&) {
+                    // failed conversion need to handle later
+                    std::exit(ERROR_CONVERSION_TO_FLOAT);
+              }
+            temp_data = match.suffix().str();
+          
+          }
+          if(count == 0){
+            //no numbers found 
+            std::exit(ERROR_NO_NUMBER_REGEX);
+          }
+          if(count > 1){
+            //error too many numbers
+            std::exit(ERROR_TOO_MANY_NUMBERS_REGEX);
+          }
+
+          close(pipe_fd[0]);
```

Comment:
I'd move this to the region immediately after the ```while ((bytes_read...)...)``` loop.

Reply:
I think it is done? What region do you mean?

Reply:
Done :+1: